controller/iteration_blackbox_test.go

```diff
@@ -636,34 +636,25 @@ func (rest *TestIterationREST) TestIterationActivatedByUser() {
 	require.Nil(rest.T(), errIdn)
 	svc, ctrl := rest.SecuredControllerWithIdentity(owner)
 	_, updated := test.UpdateIterationOK(rest.T(), svc.Context, svc, ctrl, itr1.Data.ID.String(), &payload)
-	assert.Equal(rest.T(), iteration.IterationActive, *updated.Data.Attributes.ActiveStatus) // iteration doesnot fall in timeframe yet userActive is true so iteration is active
-}
-
-// TestIterationActivatedByTimeframe tests
-// 1. Iteration should be active when it is in timeframe
-// 2. Iteration should not be active when it is outside the timeframe
-func (rest *TestIterationREST) TestIterationActivatedByTimeframe() {
-	itr1, owner := rest.createIterations()
-	assert.Equal(rest.T(), false, *itr1.Data.Attributes.UserActive)
-	assert.Equal(rest.T(), true, *itr1.Data.Attributes.ActiveStatus) // iteration falls in timeframe, so iteration is active
+	assert.Equal(rest.T(), iteration.IterationNotActive, *updated.Data.Attributes.ActiveStatus) // iteration doesnot fall in timeframe, so iteration is not active
 
-	startDate := time.Date(2017, 5, 17, 00, 00, 00, 00, time.UTC)
-	endDate := time.Date(2017, 6, 17, 00, 00, 00, 00, time.UTC)
-	payload := app.UpdateIterationPayload{
+	userActive := true
+	payload = app.UpdateIterationPayload{
 		Data: &app.Iteration{
 			Attributes: &app.IterationAttributes{
-				StartAt: &startDate,
-				EndAt:   &endDate,
+				StartAt:    &startDate,
+				EndAt:      &endDate,
+				UserActive: &userActive,
```

Comment:
Why don't you just do `payload.UserActive = &userActive`? It looks like the update payload is the same as above because you didn't change the start and end date this time.

Reply:
fixed it here e4ffadb